packages/browser/src/Session.spec.ts

```diff
@@ -902,4 +904,134 @@ describe("Session", () => {
+          .mockResolvedValue({
+            isLoggedIn: true,
+            sessionId: "a session ID",
+            webId: "https://some.webid#them",
+          });
+        const mySession = new Session({ clientAuthentication });
+        const myCallback = jest.fn((): void => {
+          expect(mySession.info.webId).toBe("https://some.webid#them");
+        });
+        mySession.events.on(EVENTS.LOGIN, myCallback);
+        await mySession.handleIncomingRedirect("https://some.url");
+        expect(myCallback).toHaveBeenCalled();
+        // Verify that the conditional assertion has been called
+        expect.assertions(2);
+      });
+    });
+
+    describe("logout", () => {
+      it("calls the registered callback on logout", async () => {
+        const myCallback = jest.fn();
+        const mySession = new Session({
+          clientAuthentication: mockClientAuthentication(),
+        });
+
+        mySession.events.on(EVENTS.LOGOUT, myCallback);
+        await mySession.logout();
+        expect(myCallback).toHaveBeenCalled();
+      });
+    });
+
+    describe("sessionRestore", () => {
+      it("calls the registered callback on session restore", async () => {
+        // Set our window's location to our test value.
+        const defaultLocation = "https://coolSite.com/resource";
+        const currentLocation = "https://coolSite.com/redirect";
+
+        // This pretends we have previously triggered silent authentication and stored
+        // the location.
+        mockLocalStorage({
+          [KEY_CURRENT_URL]: defaultLocation,
+        });
+        // This acts as the URL the user has been redirected to.
+        mockLocation(currentLocation);
+        // This pretends the login is successful.
+        const clientAuthentication = mockClientAuthentication();
+        clientAuthentication.handleIncomingRedirect = jest
+          .fn<ClientAuthentication["handleIncomingRedirect"]>()
+          .mockResolvedValue({
+            isLoggedIn: true,
+            sessionId: "a session ID",
+            webId: "https://some.webid#them",
+          });
+
+        const mySession = new Session({
+          clientAuthentication,
+        });
+        const myCallback = (urlBeforeRestore: string): void => {
+          expect(urlBeforeRestore).toEqual(defaultLocation);
+        };
+
+        mySession.events.on(EVENTS.SESSION_RESTORED, myCallback);
+        await mySession.handleIncomingRedirect(currentLocation);
+
+        // This verifies that the callback has been called
+        expect.assertions(1);
+      });
+    });
+
+    describe("sessionExpired", () => {
+      it("calls the provided callback when receiving the appropriate event", async () => {
+        const myCallback = jest.fn();
+        const mySession = new Session({
+          clientAuthentication: mockClientAuthentication(),
+        });
+        mySession.events.on(EVENTS.SESSION_EXPIRED, myCallback);
+        mySession.events.emit(EVENTS.SESSION_EXPIRED);
```

Comment:
This is a very shallow test (to say the least), which doesn't check that the event is sent at the appropriate moment. I'm questioning whether it is useful at all.

Reply:
I removed this and replaced it with an actual end-to-end test, because this unit test wasn't providing value.